qubessalt/__init__.py

```diff
@@ -93,24 +93,29 @@ def prepare_salt_config_for_vm(self):
         pillar_data = pillar_data['local']
         # remove source pillar files
         # TODO: remove also pillar modules
-        for _, roots in pillar_data['master']['pillar_roots'].items():
+        for roots in pillar_data['master']['pillar_roots'].values():
             for root in roots:
                 # do not use os.path.join on purpose - root is absolute path
                 pillar_path = tmpdir + root
                 if os.path.exists(pillar_path):
                     shutil.rmtree(tmpdir + root)
 
+        extmod_cache = pillar_data['master'].get('extension_modules',
+                                    '/var/cache/salt/minion/extmods')
+        if os.path.exists(extmod_cache):
+            shutil.copytree(extmod_cache, os.path.join(output_dir, "_extmods"))
```

Comment:
I'm a bit worried about unexpected side effects of copying dom0 modules to a VM, especially grains modules. In theory they should behave correctly, but this needs extra checking.

Reply:
Afaict all inbuilt modules have been syncing to target VMs already since their inception. They are new to the mgmt DVM only, but grain modules are not loaded by `salt-ssh` itself since it runs in a master context (it receives the grains from the target by calling `test.opts_pkg`). The `qvm_prefs` ext_pillar would be loaded if enabled in the config (I don't think this is the case).

Could you describe roughly how you imagine this extra checking/how I would implement it?